### ISSUE TITLE
Downloading error: Packaging job finished unsuccessfully for .... with a failed status

### ISSUE THREAD

**Issue by dto** [opengisch/QField#4113]
### Describe the issue
Pro QFieldCloud member here. Suddenly I cannot download or sync any QFieldCloud projects to my tablet from the cloud. After waiting several minutes, I get an error message (see attached screenshot) with "Downloading error: Packaging job finished unsuccessfully for .... with a failed status.".
![Screenshot_20230401-185337](https://user-images.githubusercontent.com/22819/229319913-58b1c31e-cbfc-4d1e-8f08-cf9148ad1976.png)

I tried creating a fresh QFIeldCloud project with the same data, still won't download to tablet.

 - QGIS Version 3.30.0
 - Mobile Device: OUKIDEL tablet
 - OS: Android 12
 - QField version: 2.7.3 Heroic Hedgehog c42fb5

My username on QFieldCloud is deeteeoh1138 and the project i'm trying to download is called NorthboroughTrails2023D. I would greatly appreciate any help you can offer.

**Comment by dto** [comment]
I tried saving my project as a GeoPackage, then did Open From... Geopackage, and converted it to a new QFieldCloud project. Then I uploaded it to QFieldCloud from QGIS Desktop, and it still failed to download to the tablet. Then , I made some small changes to the project on desktop, and re-synced with cloud. It still fails. The new project is called NorthboroughTrails2023E.

**Comment by dto** [comment]
It says there are no changes pushed. That doesn't seem right.

![Screenshot at 2023-04-01 19-29-13](https://user-images.githubusercontent.com/22819/229322684-286df95a-5548-441a-9644-d2730c53ccb0.png)

**Comment by dto** [comment]
Looks like the cloud server is failing, but no error is given from QGIS Desktop. Also the name of the project file seems messed up...

![Screenshot at 2023-04-01 19-59-00](https://user-images.githubusercontent.com/22819/229323396-0d52feb6-7959-4325-a639-3992a5ecbabd.png)
![Screenshot at 2023-04-01 19-56-31](https://user-images.githubusercontent.com/22819/229323397-8553b79b-a2b9-45d1-9bcc-964a403a1d93.png)
![Screenshot at 2023-04-01 19-56-09](https://user-images.githubusercontent.com/22819/229323398-d4388723-2e17-4ed9-88b4-c72d3347f2f4.png)
![Screenshot at 2023-04-01 19-55-48](https://user-images.githubusercontent.com/22819/229323399-c2727573-ea3c-4d74-aa7d-3904751d0d18.png)

**Comment by dto** [comment]
![Screenshot at 2023-04-01 20-08-04](https://user-images.githubusercontent.com/22819/229323616-b1df3ec1-f18e-41df-8f7e-8694465afd9b.png)
Looks like the job never finishes after creating a new project.

**Comment by dto** [comment]
I tried creating a new QGIS project file from scratch and re-importing all my layers. Then when converting it to a QFieldCloud project, it converts and uploads successfully but then gives the same error that the job failed. So I am at a complete standstill. This was working fine with a project until I made some changes on QGIS Desktop to add 2 new layers, and re-synced. Then it failed.

**Comment by dto** [comment]
I am able to manually export to QField format and load the ZIP file on the tablet. In case it might help understand what's going wrong after QFieldCloud upload, I've put the working ZIP file on my website. https://davidotoole.info/nbexport.zip
I would greatly appreciate any help you can offer. Thank you.

**Comment by dto** [comment]
I tried making a completely different project from scratch with different data sources, and it fails in the same way. My QFieldCloud account seems to be non-functional. Please let me know if you can help.

**Comment by chleeb** [comment]
I have the same problem. Downloading projects in QGIS works, but not in QField. I tested it with different accounts, devices and projects.

**Comment by Oli22** [comment]
It seems like this is a general problem with QFieldCloud, I have the same problem since yesterday on all my projects. Highly appreciate a fix or workaround, since syncing manually is no solution.

**Comment by JoaoRochaGeoExplore** [comment]
Hi. I have the exactly same project with my Qfield cloud.

Made a few changes to 2 geopackage files and start having the packaging error 
Start QGIS Application
Download Project Directory
16:49:23.635 QGSSTDERR WARNING QFSFileEngine::open: No file name specified
16:49:23.636 QGSSTDERR INFO Stopping QGIS app…

I have used the data from which I built the first project and tried it again. Can't manage to download it to my Tablet.

**Comment by dto** [comment]
@Oli22 @JoaoRochaGeoExplore @chleeb i have tried to contact the team by email (I think!) and they should be aware of it soon.

**Comment by mbernasocchi** [comment]
@Oli22 @JoaoRochaGeoExplore @chleeb @dto Thanks a lot for all your feedback we indeed noticed an increased rate in jobs failures and are investigating it. Please keep an eye on https://status.qfield.cloud as we'll be posting updates there

**Comment by mbernasocchi** [comment]
@Oli22 @JoaoRochaGeoExplore @chleeb @dto
Issue found and solved (we just had to wait a bit; nothing could have been done on our side).
The storage cluster of our cloud provider had two disk failures. The cluster rebalancing generated a lot of I/O that, in turn, generated timeouts in our workers, making some jobs unsuccessful. Thanks to the system architecture, no data were lost. If some of your jobs did not complete, you can rerun them by doing the following: 

![Screenshot from 2023-04-03 00-36-46](https://user-images.githubusercontent.com/233663/229383502-c11438ae-6311-402c-8520-ed92ff6e80e1.png)

- "Process QGIS Project File" jobs can be retriggered by reuploading your qgs/qgz file with QFieldSync 
- "Delta apply" jobs can be retriggered by reapplying changes in the changes section of qfield.cloud. 
- "Package" jobs will be automatically retriggered when you download the project in QField.

The systems are stable and we're monitoring everything.

**Comment by dto** [comment]
Hi @mbernasocchi , unfortunately this has not changed the situation for me. Both new and existing projects are still stuck at the "Process QGIS Project File" stage, which fails after a few minutes (definitely less than 10 minutes.) The error is the same: "QFSFileEngine::open: No file name specified".

**Comment by dto** [comment]
Just to verify, I cannot successfully create any new projects, even a simple minimal project with a single vector layer. They all fail at the "Process QGIS Project File" stage.

**Comment by dto** [comment]
@Oli22 @JoaoRochaGeoExplore @chleeb can you please re-check and see if you can create/sync projects now?

**Comment by Oli22** [comment]
For me it is still not working as expected on multiple projects :/

**Comment by chleeb** [comment]
I still have the problem.

**Comment by mbernasocchi** [comment]
@chleeb  @Oli22 @dto will look into it in detail since the monitoring shows a very good job succes ratio.

**Comment by mbernasocchi** [comment]
@chleeb @Oli22 @dto we've now basically 100% job passing, can you try again? some jobs might be a bi slow since the queue got a bit longer

**Comment by dto** [comment]
@mbernasocchi Hello there.  The "Process QGIS Project file" step seems to complete to the "Finished" state but there are some error messages in the log (attached)
[processlog.txt](https://github.com/opengisch/QField/files/11137340/processlog.txt)

After about 7 minutes, the "Package" job is now complete, and going to QField on my device, it now downloads and opens properly. I'll try some data recording and sync later today and let you know how it goes. So this appears to be resolved for now. Thank you. 

As for the job queue... am I correct in understanding that a slow queue on the server might result in this small project (176 MB) having processing jobs fail rather randomly because of the arbitrary 10 minute job limit? It took about 7 minutes to package, maybe 5 minutes to do "Process QGIS Project File." So I can imagine that a slow day and/or some additional layers would make this project's jobs fail. This 176 MB project is just for one town, I'll see what happens with my city project which is a bit larger but still pretty reasonable. 

Regardless, the blanket 10 minute limit is simply too restrictive. It applies equally whether one has a free account or (like me) is paying some extra euros for extra storage. I would think that the additional euros also buy some additional processing time to actually utilize the storage... that is, the price per gigabyte is inexplicable if it really does not include any additional compute time versus free accounts. Also, someone who needs 11 minutes of compute time because the system is slow that day, is likely to keep retrying N times for a total of 10*N minutes of jobs that terminate arbitrarily, instead of just using the 11 minutes he actually needed. 

Again thank you for taking the time to help me. I appreciate your assistance and I am enjoying the QField suite of products. I hope my feedback about the 10 minute limit can reach the right ears.

**Comment by idantene** [comment]
@mbernasocchi Sorry to bring this old issue up, but we're experiencing similar issues.
The QField project is quite large in nature (or well, there's a single layer that is about 600MB) - but we all have proper wifi or 5G connection, so downloading it should not be a problem.

I've been playing around the project since yesterday, but all downloads to mobile end with "Job terminated unexpectedly", whereas it works just fine on QGIS.

I'm not sure, but I suppose it could be related to the above mentioned 10 minute limit? What can be done about this?

Thanks!

**Comment by idantene** [comment]
Also pinging @opengis-ch (hopefully that's the entire team?).
I've failed to mention the above is strictly a packaging issue. I've also narrowed the file sizes and AOI so that the geopackages are under 200MiB in size, and I still reach this.
I'm not sure what exactly happens during packaging and whether or not it can be precomputed locally, for example, but I believe the 10-minute timeout is the cause here after all.
